src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_MIN_AGE = 18;
+    private static final int USER_MIN_INPUT = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws UserValidationException {
+        if (user == null) {
+            throw new UserValidationException("Null user or user parameter");
+        }
+        if (user.getLogin() == null) {
+            throw new UserValidationException("Login can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new UserValidationException("Age can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new UserValidationException("Password can't be null");
+        }
+        if (storageDao.get(user.getLogin()).equals(user)) {
```

Comment:
Using `storageDao.get(user.getLogin()).equals(user)` may lead to a `NullPointerException` if `get` returns `null` (when the user does not exist in the storage). Instead, you should check if `get` returns a non-null value before comparing it with `user`. Remember to avoid using `NullPointerException` as per the checklist.